tests/src/tests/mod.rs

```diff
@@ -200,20 +196,142 @@ async fn test_using_full_access_key() {
+
+        let expected_id = format!(
+            "{}.{factory_id}",
+            forwarder_utils::forwarder_prefix(
+                &params.target_address,
+                &params.target_network,
+                &fees_id.as_str().parse().unwrap()
+            )
+        );
+
+        assert_eq!(id.as_str(), expected_id);
     }
 }
+
+#[tokio::test]
+#[allow(clippy::similar_names)]
+async fn test_successful_complicated_flow() {
+    use crate::sandbox::factory::Factory;
+    let sandbox = Sandbox::new().await.unwrap();
+
+    let alice_address = "0x41e60a647bc61097ed52f15855fcf24a9dacdbe4";
+    let bob_address = "0xbf85c24ca42c553dbabc3ec944142e290fc3c82b";
+    let john_address = "0x78aea1a9f9880b0cda3bc0b96251948be4f340ca";
+
+    let (usdt, usdt_owner) = sandbox.deploy_ft(TOTAL_SUPPLY, "USDT", 6).await.unwrap();
+    let (usdc, usdc_owner) = sandbox.deploy_ft(TOTAL_SUPPLY, "USDC", 6).await.unwrap();
+
+    let silo1 = sandbox.deploy_aurora("silo-1").await.unwrap();
+    let silo2 = sandbox.deploy_aurora("silo-2").await.unwrap();
+    let silo3 = sandbox.deploy_aurora("silo-3").await.unwrap();
+
+    usdt.storage_deposit(silo1.id()).await.unwrap();
+    usdc.storage_deposit(silo1.id()).await.unwrap();
+    usdt.storage_deposit(silo2.id()).await.unwrap();
+    usdc.storage_deposit(silo2.id()).await.unwrap();
+    usdt.storage_deposit(silo3.id()).await.unwrap();
+    usdc.storage_deposit(silo3.id()).await.unwrap();
+
+    let usdt_erc20_s1 = silo1.deploy_erc20(usdt.id()).await.unwrap();
+    let usdc_erc20_s1 = silo1.deploy_erc20(usdc.id()).await.unwrap();
+    let usdt_erc20_s2 = silo2.deploy_erc20(usdt.id()).await.unwrap();
+    let usdc_erc20_s2 = silo2.deploy_erc20(usdc.id()).await.unwrap();
+    let usdt_erc20_s3 = silo3.deploy_erc20(usdt.id()).await.unwrap();
+    let usdc_erc20_s3 = silo3.deploy_erc20(usdc.id()).await.unwrap();
+
+    let fees = sandbox.deploy_fees(&[usdt.id(), usdc.id()]).await.unwrap();
+
+    usdt.storage_deposit(fees.id()).await.unwrap();
+    usdc.storage_deposit(fees.id()).await.unwrap();
+
+    let factory = sandbox.deploy_factory(fees.id()).await.unwrap();
+
+    let parameters = [
+        DeployParameters {
+            target_address: alice_address.to_string(),
+            target_network: silo1.id().as_str().parse().unwrap(),
+        },
+        DeployParameters {
+            target_address: bob_address.to_string(),
+            target_network: silo2.id().as_str().parse().unwrap(),
+        },
+        DeployParameters {
+            target_address: john_address.to_string(),
+            target_network: silo3.id().as_str().parse().unwrap(),
+        },
+    ];
+    let forward_ids = factory.create(&parameters).await.unwrap();
+
+    for fwd_id in &forward_ids {
+        usdt.storage_deposit(fwd_id).await.unwrap();
+        usdc.storage_deposit(fwd_id).await.unwrap();
+    }
+
+    let [ref alice_fwd, ref bob_fwd, ref john_fwd] = forward_ids[..] else {
+        panic!("")
+    };
```

Comment:
Nit: this conditional structural match is a little messy. I would re-write it in a way with better compile-time enforcment (relating the length of `parameters` to the length of `forward_ids`):

```diff
diff --git a/tests/src/tests/mod.rs b/tests/src/tests/mod.rs
index 992b4bc..56ef48a 100644
--- a/tests/src/tests/mod.rs
+++ b/tests/src/tests/mod.rs
@@ -233,6 +233,7 @@ async fn test_using_factory() {
 #[allow(clippy::similar_names)]
 async fn test_successful_complicated_flow() {
     use crate::sandbox::factory::Factory;
+    const N: usize = 3;
     let sandbox = Sandbox::new().await.unwrap();

     let alice_address = "0x41e60a647bc61097ed52f15855fcf24a9dacdbe4";
@@ -267,7 +268,7 @@ async fn test_successful_complicated_flow() {

     let factory = sandbox.deploy_factory(fees.id()).await.unwrap();

-    let parameters = [
+    let parameters: [DeployParameters; N] = [
         DeployParameters {
             target_address: alice_address.to_string(),
             target_network: silo1.id().as_str().parse().unwrap(),
@@ -281,16 +282,19 @@ async fn test_successful_complicated_flow() {
             target_network: silo3.id().as_str().parse().unwrap(),
         },
     ];
-    let forward_ids = factory.create(&parameters).await.unwrap();
+    let forward_ids: [_; N] = factory
+        .create(&parameters)
+        .await
+        .unwrap()
+        .try_into()
+        .unwrap();

     for fwd_id in &forward_ids {
         usdt.storage_deposit(fwd_id).await.unwrap();
         usdc.storage_deposit(fwd_id).await.unwrap();
     }

-    let [ref alice_fwd, ref bob_fwd, ref john_fwd] = forward_ids[..] else {
-        panic!("")
-    };
+    let [alice_fwd, bob_fwd, john_fwd] = &forward_ids;

     usdt.ft_transfer(&usdt_owner, alice_fwd, 8_000_000)
         .await
```